operator/validator/event_handler.go

```diff
@@ -123,6 +142,53 @@ func (c *controller) handleValidatorAddedEvent(
 	event abiparser.ValidatorAddedEvent,
 	ongoingSync bool,
 ) ([]zap.Field, error) {
+	_, found, err := c.eventHandler.GetEventData(event.TxHash)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get event data")
+	}
+	if found {
+		// skip
+		return nil, nil
+	}
+
+	// get nonce
+	nonce, err := c.eventHandler.GetNextNonce(event.Owner)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get nonce")
+	}
+
+	err = c.eventHandler.SaveEventData(event.TxHash)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to save event data")
+	}
+
+	err = c.eventHandler.BumpNonce(event.Owner)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to bump the nonce")
+	}
+
+	// Calculate the expected length of constructed shares based on the number of operator IDs,
+	// signature length, public key length, and encrypted key length.
+	operatorCount := len(event.OperatorIds)
+	signatureOffset := phase0.SignatureLength
+	pubKeysOffset := phase0.PublicKeyLength*operatorCount + signatureOffset
+	sharesExpectedLength := encryptedKeyLength*operatorCount + pubKeysOffset
+
+	if sharesExpectedLength != len(event.Shares) {
+		return nil, errors.Errorf("%s event shares length is not correct", abiparser.ValidatorAdded)
+	}
```

Comment:
The error should be follows
```
eturn nil, &MalformedEventError{
	Err: errors.Wrap(err, "failed to verify signature"),
}
```

Reply:
In general we have to ditignuish between Malformed and sytem errors,
and increment the Nonce only for malformed.
This way we can ensure the validity of the Nonce